### PR TITLE
Remove dependency on track 1 Storage mgmt lib

### LINKED ISSUE #11088: Create storage account in test-resources.json and update docs to tell people how to use it
Rather than take a dependency on track 1 management libraries to create a storage account, use test-resources.json to create a shared one, much like we do for a resource group. We should then create randomized containers for tests as needed (see referenced code) and/or just have a shared one since the DataSource should probably be immutable. This would save time and resources as well.

We must also update the documentation (like CONTRIBUTING.md) to tell people how to make use of this to set up their F5 and test run scenarios, e.g. run eng/common/New-TestResources.ps1. This should be used to create both the resource group ("rg-$BaseName") as well as the storage account. This should simplify instructions, rather than have them create an RG and then run this to create the storage account.

> I would just prefer to avoid management plane dependencies in our tests.  We couldn't for the Search service instances.  If it's possible to do that for Storage, I would prefer to and maintain as much adherence to the unified test-resources.json approach as possible.  We'd have to create one ahead of time for local testing, but we already have to do that for the resource group/principal/etc.  We'd just have another env var with the connection string.
>
> _Originally posted by @tg-msft in https://github.com/Azure/azure-sdk-for-net/pull/11049/files/997aa4b64a88d4b71e56aa26ff52614d3acef52c..aa953217b6831eb42175c072f66caa3ff9cd58ec#r404339396_

### PR DESCRIPTION
**Pull request by heaths**
Changes CONTRIBUTING.md to use the New-TestResources.ps1 script, along with related PowerShell cmdlets you'd have installed to use the script (instead of mixing that with the az CLI). Removes the dependency on track 1 Microsoft.Azure.Management.Search.

The accounts used to record were removed immediately after successfully recording and playing back tests.

Fixes #11088

### CODE REVIEW

File: eng/common/TestResources/New-TestResources.ps1 [hunk 1]
```diff
@@ -109,7 +109,7 @@ trap {
 }
 
 # Enumerate test resources to deploy. Fail if none found.
-$root = [System.IO.Path]::Combine("$PSScriptRoot/../sdk", $ServiceDirectory) | Resolve-Path
+$root = [System.IO.Path]::Combine("$PSScriptRoot/../../../sdk", $ServiceDirectory) | Resolve-Path
```
Comment (heaths) [inline eng/common/TestResources/New-TestResources.ps1:112]: This was already fixed in azure-sdk-tools and merged into the repo. I forgot to rebase this on master but will effectively no-op when merged (not a conflict).

File: sdk/search/CONTRIBUTING.md [hunk 2]
```diff
@@ -72,57 +72,54 @@ live tests against a pull request by commenting `/azp run net - search - tests`
 in the PR.
 
 ### Live Test Resources
-You'll need to pass the details for a subscription, service principal, resource
-group, and location via environment variables.  We set this up for live tests
-using the [test-resources.json](./test-resources.json) ARM template, but it's
-equivalent to the following steps if you want to run the live tests locally.
+To set up your Azure account to run live tests, you'll need to log into Azure,
```
Comment (heaths) [inline sdk/search/CONTRIBUTING.md:75]: Should we add a blurb - perhaps even a link - to first install the Azure module? We didn't actually document to install the az CLI, but I'm open to adding something if this is to become our template for other services.

Comment (pakrym) [reply sdk/search/CONTRIBUTING.md:75]: Do we need these instructions in every service directory that uses live resource management? Can we centralize them and link to a central location?

File: sdk/search/CONTRIBUTING.md [hunk 3]
```diff
@@ -72,57 +72,54 @@ live tests against a pull request by commenting `/azp run net - search - tests`
...
+  -TestApplicationSecret (ConvertFrom-SecureString $sp.Secret -AsPlainText)
+```
+
+Along with some log messages, this will output environment variables based on your
+current shell like in the following example:
+
+```
+$env:AZURE_TENANT_ID = '04acef35-c7bd-4d14-bfd9-59f11b7b9eac'
+$env:AZURE_CLIENT_ID = 'ce1a3a01-424f-4e34-b9a6-823e2b1ae783'
+$env:AZURE_CLIENT_SECRET = 'c27ccc92-e1ca-4d29-8f4e-c6a1ee61ff57'
```
Comment (weshaggard) [inline sdk/search/CONTRIBUTING.md:97]: I hope these are invalid secrets.

File: sdk/search/CONTRIBUTING.md (hunk 2 above)
Comment (weshaggard) [reply sdk/search/CONTRIBUTING.md:75]: Yes I think we should be in the central location with a link and perhaps one example command line that is specialized for the service, but if it is only the service directory maybe that isn't very interesting.

File: sdk/search/CONTRIBUTING.md (hunk 3 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:97]: Of course. 🙂

File: sdk/search/CONTRIBUTING.md (hunk 2 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:75]: I mostly agree, but the output will be different for some services. I could make a comment to that affect - that people should copy the whole output and that the example is just that - example output that may vary (I'll probably just use this exactly as an example).

Actually, one other concern: our docs are a mess. Should this be an absolute URL to a central doc - perhaps a section in /CONTRIBUTING.md. Agreed?

File: sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json [hunk 4]
```diff
@@ -19,59 +19,59 @@
         "x-ms-return-client-request-id": "true"
       },
       "RequestBody": {
-        "name": "azsnetyajaxiuy",
+        "name": "adpnetsearchstg",
         "type": "azureblob",
         "credentials": {
-          "connectionString": "DefaultEndpointsProtocol=https;AccountName=azsnetyajaxiuy;AccountKey=H4Q6\u002BwLogzoaDNWc1BLxgE7OkztuFjblhsRC\u002BKARuaAMt1T1xgdtOvxSG72LVYn9GH0WO65M4eHTMB6h2vQWFA==;EndpointSuffix=core.windows.net"
+          "connectionString": "DefaultEndpointsProtocol=https;AccountName=adpnetsearchstg;AccountKey=r72v9Brt0bNoOiXHv0wQ9EPNP52SlSxCxspjwfOYFTp\u002BzlMUFURgwUp0hNUHE/Vwnx8UZZfx4hh1n9i0zpIRzA==;EndpointSuffix=core.windows.net"
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json:25]: This should be sanitized even if it's (usually) deleted by the time recordings get checked in.

File: sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs [hunk 5]
```diff
@@ -554,63 +528,22 @@ private async Task<SearchResources> CreateSearchServiceIndexAndDocumentsAsync()
...
-        /// <returns>The current <see cref="SearchResources"/>.</returns>
-        private async Task<SearchResources> CreateHotelsContainerAsync()
-        {
-            if (TestFixture.Mode != RecordedTestMode.Playback)
-            {
                 using CancellationTokenSource cts = new CancellationTokenSource(Settings.Timeout);
 
-                BlobContainerClient client = new BlobContainerClient(StorageAccountConnectionString, "hotels");
+                BlobContainerClient client = new BlobContainerClient(StorageAccountConnectionString, BlobContainerName);
                 await client.CreateAsync(cancellationToken: cts.Token);
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs:543]: Should probably make this `CreateIfNotExistsAsync` instead.  It's safer for edge cases where a network failure causes a retry even though the service received and responded.

File: sdk/search/CONTRIBUTING.md [hunk 6]
```diff
@@ -72,57 +72,54 @@ live tests against a pull request by commenting `/azp run net - search - tests`
...
-
-PS C:\src> setx AZURE_SUBSCRIPTION_ID <subscription_id>
-SUCCESS: Specified value was saved.
-
-PS C:\src> setx AZURE_RESOURCE_GROUP <resource_group>
-SUCCESS: Specified value was saved.
-
-PS C:\src> setx AZURE_LOCATION <location>
-SUCCESS: Specified value was saved.
+PS C:\src> Connect-AzAccount -Subscription 'YOUR SUBSCRIPTION ID'
```
Comment (tg-msft) [inline sdk/search/CONTRIBUTING.md:82]: Are the PS cmdlets required for this?  We've preferred using the CLI for all of our samples.

File: sdk/search/CONTRIBUTING.md [hunk 7]
```diff
@@ -72,57 +72,54 @@ live tests against a pull request by commenting `/azp run net - search - tests`
...
+$env:AZURE_TENANT_ID = '04acef35-c7bd-4d14-bfd9-59f11b7b9eac'
+$env:AZURE_CLIENT_ID = 'ce1a3a01-424f-4e34-b9a6-823e2b1ae783'
+$env:AZURE_CLIENT_SECRET = 'c27ccc92-e1ca-4d29-8f4e-c6a1ee61ff57'
+$env:AZURE_SUBSCRIPTION_ID = 'd686c17c-aade-4238-bce0-290453cfcf97'
+$env:AZURE_RESOURCE_GROUP = 'rg-myusername'
+$env:AZURE_LOCATION = 'westus2'
+$env:AZURE_SEARCH_STORAGE_NAME = 'myusernamestg'
+$env:AZURE_SEARCH_STORAGE_KEY = 'Of2O5Snep5tl13bfjh02/fSNYfrBPXV7CYK7EVnMm/z9fN7zCcq6WKuWfZDM9QsTORvC7zYLifyIEtymI5VCmA=='
+
+dir env:AZURE* | % { setx $_.Name $_.Value }
```
Comment (tg-msft) [inline sdk/search/CONTRIBUTING.md:122]: Let's not do this.  I don't want to give users commands that might cause configuration weirdness.  I'd prefer to just give them a list of setx commands to paste.

File: sdk/search/CONTRIBUTING.md [hunk 8]
```diff
@@ -72,57 +72,54 @@ live tests against a pull request by commenting `/azp run net - search - tests`
...
+
+For security reasons we do not set these environment variables automatically for either
+the current process or persistently for future sessions. You must do that yourself.
+
+If your current shell was detected propertly, you should be able to copy and paste the
+output directly in your terminal and add to your profile script. For example,
+in PowerShell on Windows you could copy and paste the output and run the following command:
+
+```powershell
+$env:AZURE_TENANT_ID = '04acef35-c7bd-4d14-bfd9-59f11b7b9eac'
```
Comment (tg-msft) [inline sdk/search/CONTRIBUTING.md:113]: Nit - above we were using a `PS C:\src>` prompt

**Review by tg-msft — APPROVED** [review]

File: sdk/search/CONTRIBUTING.md (hunk 6 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:82]: That's changing with the new system we're starting to roll out. I'll be linking to a central doc (offline conversation) and will include a blurb about installing them.

File: sdk/search/CONTRIBUTING.md (hunk 7 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:122]: This was decided for security reasons. The scripts don't actually set the environment variables, and it has to be cross-platform. setx is Windows. On other platforms, persisting variables across sessions (like in .bashrc, .profile, etc.) varies from shell to shell, or even with customizations. That's another reason we don't do it.

File: sdk/search/CONTRIBUTING.md (hunk 8 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:113]: Which makes it impossible to simply copy and paste. Doing so here executes them in sequence and just works. It also made sense before because input and output was mixed. That's not the case here.

File: sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json (hunk 4 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json:25]: The account name and primary key must be persisted, as you can see in variables below or playback can't work. There would be no point sanitizing these variables since they have to appear below. They were deleted (and the primary key token will never be the same, even if the account name is recreated) before I ever pushed a commit with these changes.

File: sdk/search/CONTRIBUTING.md (hunk 7 above)
Comment (tg-msft) [reply sdk/search/CONTRIBUTING.md:122]: I'm not talking about the script.  I'm talking about your example here.  It's fine to have a bunch of `setx` examples.  I don't think we should `setx` every AZURE env var in your current process.  I think this .md file should delete line 122.

File: sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json (hunk 4 above)
Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json:25]: I know _you_ deleted these.  Somebody else won't in the future when they create resources locally.  We should have a Sanitizer strip the key out.  That's how all of the Storage tests work.

File: sdk/search/CONTRIBUTING.md (hunk 7 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:122]: Then where do you suggest people persist them? If they lose them, they are gone forever. Remember, you wanted me to change the instructions for all persistent live resources. Creating them on the fly is for CIs/nightlies - not dev testing. Those resources are meant to be persisted, as the old instructions described.

Comment (tg-msft) [reply sdk/search/CONTRIBUTING.md:122]: I'm fine with doing `setx` on the explicit set of variables we create.  I'm saying we shouldn't `setx` everything on a random user's machine that starts with AZURE.

File: sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json (hunk 4 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json:25]: Only if they record, in which case they have to get stored as variables as well. If they push commits with those recordings, they will have to remove them. Sanitizing these specific request/response lines won't change the fact they *must* be in variables. This is a result of putting the storage account creation in the ARM template. This stuff was created before hand, so it needs to get persisted for playback (even the resources were removed).

File: sdk/search/CONTRIBUTING.md (hunk 6 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:82]: Part of the problem was because of automating acquisition and cross-platform scripting of the azure CLI turned up some issues. Using the azure CLI was my original intent, but to have a cross-platform scripting interface - pwsh - was needed anyway, acquisition of the azure CLI was not only way slower (seems to keep a copy of python in it, and tons of tiny files take forever to extract), but deemed extraneous when we can just get the module from within PowerShell we have to use already. This was a cross-team decision.

File: sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json (hunk 4 above)
Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json:25]: We should just sanitize the `AccountKey` part of the connection string.  The rest of the values will be there to keep playback the same.  Then nobody can ever accidentally mess this up.

File: sdk/search/CONTRIBUTING.md (hunk 2 above)
Comment (heaths) [reply sdk/search/CONTRIBUTING.md:75]: Discussed offline. I'll add a README.md to eng/common/TestResources (I'll sync up to azure-sdk-tools after we agree on content with this PR) and point to there from this README.md, and also link from the repo root's CONTRIBUTING.md.

/cc @danieljurek

File: sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json (hunk 4 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/SessionRecords/SearchServiceClientTests/CreateAzureBlobIndexer.json:25]: Done. Right now, if the content is application/json I am allocating a `StringBuilder` each time which I did verify copies content. Since the lifetime of the test class is all methods for NUnit, this affects all methods. We could optimize it out at some point if this grows, or I have an idea for a helper struct (even fewer allocations) that will conditionally allocate a `StringBuilder` and either return it or the original string if we feel it necessary for tests when recording. I just didn't want to add it if no one was interested in this particular scenario.

**Review by danieljurek — APPROVED** [review]
Documentation changes LGTM, one small suggestion.

File: eng/common/TestResources/README.md [hunk 9]
```diff
@@ -1,29 +1,112 @@
...
+
+```powershell
+$env:AZURE_TENANT_ID = '04acef35-c7bd-4d14-bfd9-59f11b7b9eac'
+$env:AZURE_CLIENT_ID = 'ce1a3a01-424f-4e34-b9a6-823e2b1ae783'
+$env:AZURE_CLIENT_SECRET = 'c27ccc92-e1ca-4d29-8f4e-c6a1ee61ff57'
+$env:AZURE_SUBSCRIPTION_ID = 'd686c17c-aade-4238-bce0-290453cfcf97'
+$env:AZURE_RESOURCE_GROUP = 'rg-myusername'
+$env:AZURE_LOCATION = 'westus2'
+$env:AZURE_SEARCH_STORAGE_NAME = 'myusernamestg'
+$env:AZURE_SEARCH_STORAGE_KEY = 'Of2O5Snep5tl13bfjh02/fSNYfrBPXV7CYK7EVnMm/z9fN7zCcq6WKuWfZDM9QsTORvC7zYLifyIEtymI5VCmA=='
```
[the remark this answers is not in the document]
Comment (heaths) [reply eng/common/TestResources/README.md:44]: Good point. Just the storage key, or the GUIDs as well? As much as practical, I'd like to keep the example as close to the real output as possible to cause less confusion.

[the remark this answers is not in the document]
Comment (weshaggard) [reply eng/common/TestResources/README.md:44]: I think it is probably OK to just put in a placeholder here as folks aren't going to be copying and pasting this.

File: sdk/search/Azure.Search.Documents/tests/Utilities/SearchRecordedTestSanitizer.cs [hunk 10]
```diff
@@ -29,5 +61,26 @@ public override void SanitizeHeaders(IDictionary<string, string[]> headers)
...
+        public override string SanitizeVariable(string variableName, string environmentVariableValue)
+        {
+            if (SearchResources.StorageAccountKeyVariableName.Equals(variableName, StringComparison.OrdinalIgnoreCase))
+            {
+                // Assumes the secret content is destined to appear in JSON, for which certain common characters in account keys are escaped.
+                // See https://github.com/dotnet/runtime/blob/8640eed0/src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs
+                string encoded = JavaScriptEncoder.Default.Encode(environmentVariableValue);
+                _secrets.Add(encoded);
+
+                return SanitizeValue;
```
Comment (tg-msft) [inline sdk/search/Azure.Search.Documents/tests/Utilities/SearchRecordedTestSanitizer.cs:80]: So this replaces the entire connection string with "SANITIZED"?  I suppose if we're not actually uploading anything (and consequently instrumenting our BlobContainerClient, etc.) then it's fine.

**Review by weshaggard — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/tests/Utilities/SearchRecordedTestSanitizer.cs (hunk 10 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/Utilities/SearchRecordedTestSanitizer.cs:80]: No, just the AccountKey. See `SanitizeBodyText` above. I suppose it could sanitize the whole thing, but the rest doesn't contain anything not already exposed (and okay to do so), but I don't see any benefit to push another commit.

Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/tests/Utilities/SearchRecordedTestSanitizer.cs:80]: Oh, sorry, I thought we had a connection string.  Makes sense now.

**Comment by heaths** [comment]
Something strange going on with Windows_NetFramework build. Re-kicking.

/azp run net - core - ci
